File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    void findSubsetSum(int i, int[] nums, int N, ArrayList<Integer> arr, Set<List<Integer>> ans){
        if(i>=N){
            ans.add(new ArrayList<>(arr));
            return ;
        }
        arr.add(nums[i]);
        findSubsetSum(i+1,nums,N,arr,ans);
        arr.remove(arr.size()-1);
        findSubsetSum(i+1,nums,N,arr,ans);
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        int N=nums.length;Arrays.sort(nums);
        ArrayList<Integer> arr=new ArrayList<Integer>();
        Set<List<Integer>> ans=new HashSet<>();
        findSubsetSum(0,nums,N,arr,ans);
        List<List<Integer>> s=new  ArrayList<>();
        s.addAll(ans);
        return s;
    }
}

File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
    void helper(int i, vector<int>& nums, set<vector<int>> &ans,  vector<int> &temp){
        if(i>=nums.size()){
            ans.insert(temp);
            return;
        }
        temp.push_back(nums[i]);
        helper(i+1,nums,ans,temp);
        temp.pop_back();
         helper(i+1,nums,ans,temp);
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        set<vector<int>> ans;
        vector<int> temp;
        sort(nums.begin(),nums.end());
        helper(0,nums,ans,temp);
        vector<vector<int>> res(ans.begin(),ans.end());
        return res;
    }
};